tests/ert3/console/integration/test_cli.py

```diff
@@ -121,3 +122,132 @@ def test_cli_record_load_not_existing_file(tmpdir):
+        if idx in done_indices:
+            assert experiment in done
+        else:
+            assert experiment in pending
+
+
+def test_cli_status_no_runs(tmpdir, capsys):
+    workspace = tmpdir / _POLY_WORKSPACE_NAME
+    shutil.copytree(_POLY_WORKSPACE, workspace)
+    workspace.chdir()
+
+    experiments = ert3.workspace.get_experiment_names(workspace)
+
+    args = ["ert3", "init"]
+    with patch.object(sys, "argv", args):
+        ert3.console.main()
+
+    args = ["ert3", "status"]
+    with patch.object(sys, "argv", args):
+        ert3.console.main()
+
+    _assert_done_or_pending(capsys.readouterr(), experiments, [])
+
+
+def test_cli_status_some_runs(tmpdir, capsys):
+    workspace = tmpdir / _POLY_WORKSPACE_NAME
+    shutil.copytree(_POLY_WORKSPACE, workspace)
+    workspace.chdir()
+
+    experiments = list(ert3.workspace.get_experiment_names(workspace))
+
+    args = ["ert3", "init"]
+    with patch.object(sys, "argv", args):
+        ert3.console.main()
+
+    done_indices = [1, 3]
+    for idx in done_indices:
+        ert3.storage.init_experiment(
+            workspace=workspace, experiment_name=experiments[idx], parameters=[]
+        )
+
+    args = ["ert3", "status"]
+    with patch.object(sys, "argv", args):
+        ert3.console.main()
+
+    _assert_done_or_pending(capsys.readouterr(), experiments, [1, 3])
+
+
+def test_cli_status_all_run(tmpdir, capsys):
+    workspace = tmpdir / _POLY_WORKSPACE_NAME
+    shutil.copytree(_POLY_WORKSPACE, workspace)
+    workspace.chdir()
+
+    experiments = ert3.workspace.get_experiment_names(workspace)
+
+    args = ["ert3", "init"]
+    with patch.object(sys, "argv", args):
+        ert3.console.main()
+
+    for experiment in experiments:
+        ert3.storage.init_experiment(
+            workspace=workspace, experiment_name=experiment, parameters=[]
+        )
+
+    args = ["ert3", "status"]
+    with patch.object(sys, "argv", args):
+        ert3.console.main()
+
+    _assert_done_or_pending(capsys.readouterr(), experiments, range(len(experiments)))
+
+
+def test_cli_status_no_experiments(tmpdir, capsys):
+    workspace = tmpdir
+    workspace.chdir()
+    workspace.mkdir(ert3.workspace.EXPERIMENTS_BASE)
+
+    args = ["ert3", "init"]
+    with patch.object(sys, "argv", args):
+        ert3.console.main()
+
+    args = ["ert3", "status"]
+    with patch.object(sys, "argv", args):
+        ert3.console.main()
+
+    captured = capsys.readouterr()
+    assert captured.out.strip() == "No experiments present in this workspace"
+
+
+def test_cli_status_no_experiments_root(tmpdir):
+    workspace = tmpdir
+    workspace.chdir()
+    workspace.mkdir(ert3.workspace.EXPERIMENTS_BASE)
+
+    args = ["ert3", "init"]
+    with patch.object(sys, "argv", args):
+        ert3.console.main()
+
+    shutil.rmtree(ert3.workspace.EXPERIMENTS_BASE)
+
+    args = ["ert3", "status"]
+    with patch.object(sys, "argv", args):
+        with pytest.raises(SystemExit) as error:
```

Comment:
Should be an `IllegalWorkspaceState` exception I presume?

Reply:
No, that did not work. The older tests you wrote do the same, it seems that we use a function that catches exceptions and  does an exit. Probably easily fixed, will look into it, because we should be checking for the right exceptions in a good test.

Reply:
Ah, yes. The cli catches the ERT exceptions to present them in a neat manner to the user... If we want to test towards this exception you would have to go directly towards the engine...